File: src/main/java/com/shadowbring/todo/domain/ToDoItem.java
package com.shadowbring.todo.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

@Entity
@Data
@EqualsAndHashCode(exclude = "id")
@SequenceGenerator(name = "itemSequence", sequenceName = "itemSequence", allocationSize = 1)
public class ToDoItem {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "itemSequence")
  private Long id;

  @Column(nullable = false)
  private String description;

  @Column(nullable = false)
  private Boolean done;

  @ManyToOne(optional = false)
  @JoinColumn(name = "toDoUserId")
  private ToDoUser toDoUser;
}

File: src/main/java/com/shadowbring/todo/domain/ToDoItem.kt
package com.shadowbring.todo.domain

import javax.persistence.*

@Entity
@SequenceGenerator(name = "itemSequence", sequenceName = "itemSequence", allocationSize = 1)
data class ToDoItem(

        @Id
        @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "itemSequence")
        private var id: Long = 0,

        @Column(nullable = false)
        var description: String = "",

        @Column(nullable = false)
        var done: Boolean = false,

        @ManyToOne(optional = false)
        @JoinColumn(name = "toDoUserId")
        private var toDoUser: ToDoUser = ToDoUser()
)